Formula/spack.rb

```diff
@@ -0,0 +1,21 @@
+class Spack < Formula
+  desc "Package manager that builds multiple versions and configurations of software"
+  homepage "https://spack.io"
+  url "https://github.com/spack/spack/releases/download/v0.16.0/spack-0.16.0.tar.gz"
+  sha256 "064b2532c70916c7684d4c7c973416ac32dd2ea15f5c392654c75258bfc8c6c2"
+  license any_of: ["Apache-2.0", "MIT"]
+
+  head "https://github.com/spack/spack.git", branch: "develop"
+
+  bottle :unneeded
+
+  depends_on "python"
+
+  def install
+    cp_r Dir["bin", "etc", "lib", "share", "var"], prefix.to_s
+  end
+
+  test do
+    system "#{bin}/spack", "--version"
```

Comment:
This test doesn't look sufficent. Cf. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

>We want tests that don’t require any user input and test the basic functionality of the application. For example foo build-foo input.foo is a good test and (despite their widespread use) foo --version and foo --help are bad tests.

Reply:
I added in a test with `spack list zlib` which mirrors the vcpkg test `vcpkg search sqlite` -- it doesn't depend on being able to download zlib files.

When running tests, does brew prevent file system operations or somehow restrict them?

Reply:
Yes, there is some restriction on what you can do, as the tests are run in a sandbox. Is the new test giving you errors?

Reply:
`spack list` works with the restriction, `spack install` was giving an error when it tries to write files.

Reply:
Yes, you are restricted on where you can write. If you wish to test `spack install` (I think it would be useful to), make sure it writes to `testpath`. Maybe something like `system "spack", "install", testpath` or something like that would work. (Well, modulo specifying what it is you want to install.)

Reply:
I think I got a working test that installs a package sorted out. It ended up being an adventure writing out a config file that changes several paths to folders in `testpath` and then picking up enough Ruby to convert a string with the zlib install prefix to a Pathname that can be used with `:exist?`.

Reply:
Yea, I know how that feels. I only really picked up Ruby when I started contributing to Homebrew. The test is looking very good though.